lipx.py

```diff
@@ -45,6 +35,17 @@ def get_uint24(data, index):
     return int((data[index] << 16) | (data[index + 1] << 8) | data[index + 2])
 
 
+# Helper function to display patch data in INSPECT mode
+def format_patch(offset, size, data, rle=0):
+    if (offset ^ size) & 1:
+        data.append(0)
+        data = data.hex(' ', 2)[:-2]
+    else:
+        data = data.hex(' ', 2)
+    repeat = f' REPEAT x {rle}' if rle else ''
+    return f'{hex(offset)[2:].zfill(6)} ({str(size).zfill(2)}): {"  " * (offset & 1) + data}{repeat}'
```

Comment:
i would prefer a more consistent output to improve readability and avoid this zig-zag

![Screenshot_20230711_200940](https://github.com/kylon/Lipx/assets/3252255/3db12b41-6919-4ddb-9f65-704aaa4f81b5)

Reply:
@kylon I did this deliberately so that odd and even byte offsets would line up with how the full file was displayed in a hex dump or other hex editor.

Maybe I haven't described it very well, but the idea is that odd byte offsets have a whitespace pad to keep the word boundaries aligned when viewing the full binary file in another hex editor. 

```
000002 (04): hhhh hhhh __
000009 (04): __hh hhhh hh
```

for example, this is the alignment using command line `xxd`:

```
00005f70: 0000 0000 0000 0000 0000 3c7e 7e66 607c  ..........<~~f`|
00005f80: 0000 0000 0000 0000 7e66 667e 7e3c 0000  ........~ff~~<..
00005f90: 0000 0000 0000 0000 0000 7e7e 7e66 0e0c  ..........~~~f..
00005fa0: 0000 0000 0000 0000 1c18 1818 1818 0000  ................
```

In your screenshot the last line looks wrong though, I'd expect it to be:

    0038a9 (03): __20 9afe

to keep the odd/even alignment I intended. Looks like there is a bug there and that's not what I intented, I'll investigate.

Does what I'm trying to do with the odd/even byte alignment make sense though? I found it useful when comparing the patch output to the full file hex for the 2 byte word groups to line up.

Reply:
ok, fine for me, maybe just add a note in the output about how data is displayed
thanks